### ISSUE TITLE
Can't entirely disable ignore file discovery

### ISSUE THREAD

**Issue by paulirish** [watchexec/watchexec#643]
This is minor but maybe relevant to the [Ignore handling issues Milestone](https://github.com/watchexec/watchexec/milestone/10)...

Over in https://github.com/watchexec/watchexec/issues/421 , a while ago, I asked... 

> Perhaps there's a way to serve fd's output to watchexec so I can skip over its own ignore discovery?

And passcod replied: 

> sure, yeah, write a filter file with it, disable ignores

I've now tried that and it looks like watchexec is still doing ignorefile discovery (that's fairly costly)..

```sh
fd -j1 --type file > .trackedfiles  # fd == fdfind on linux
watchexec --filter-file .trackedfiles --no-default-ignore   --no-global-ignore --no-project-ignore  --no-vcs-ignore --project-origin . -vvv -- date
```

I see 30 seconds before the `watchexec_cli::filterer::common: discovered ignore files from project origin` line. And that line has _lots_ of entries. (The Chromium repo is a monster)

- Watchexec's version: `watchexec 1.22.3 (2023-05-14)`
- The OS you're using: Mac OS Ventura 13.5
- A log with `-vvv --log-file`: https://gist.github.com/paulirish/1f6d399d3120253b0bceeb5db2fcdce3
- A sample command: see above

**Comment by passcod** [comment]
Oh yep, I can fix that up

**Comment by passcod** [comment]
Released in 1.23.0. I've also added `--no-discover-ignore` which absorbs `--no-global-ignore --no-project-ignore --no-vcs-ignore` and skips even more discovery logic.
